rails-ubuntu-simple/README.md

```diff
@@ -0,0 +1,239 @@
+convenient base image to build from. The ruby postgres client gem
+depends on libpq-dev being installed, so we'll also to install that as
+part of our Dockerfile.
+
+```
+FROM rails:onbuild
+RUN apt-get update -qq && apt-get install -y build-essential libpq-dev
+```
+
+With our Dockerfile configured, we are ready to build the container
+image. As usual with a docker container, if you change your project
+you will need to re-build the image before running your container.
+
+```bash
+$ docker build -t webapp .
+```
+
+For more information about dockerizing rails applications, see [the
+rails docker image
+documentation](https://registry.hub.docker.com/_/rails/)
+
+## Launching Weave ##
+
+Launch weave, with weaveDNS. Weave will be responsible for the raw
+network connection and routing between our containers.
+[WeaveDNS](http://docs.weave.works/weave/latest_release/weavedns.html)
+will provide service-discovery between the app, and the database
+container.
+
+```bash
+$ weave launch && weave launch-dns
+```
+
+In this guide we'll configure our docker client to use the [weave
+proxy](http://docs.weave.works/weave/latest_release/proxy.html). This
+lets us use the standard docker client, and will attach any booted
+containers to the weave network. We'll launch the proxy and then use
+the proxy-env utility to configure our DOCKER_HOST.
+
+```bash
+$ weave launch-proxy
+$ eval $(weave proxy-env)
+```
+
+## Launching PostgreSQL ##
+
+Now, launching a PostgreSQL instance with DNS service discovery can be
+done with one command. For convenience, we'll use the standard
+[postgres image from dockerhub](https://registry.hub.docker.com/_/postgres/).
+
+```bash
+$ docker run --name db -e POSTGRES_PASSWORD=mysecretpassword -d postgres
+```
+
+By setting the container name, our postgres instance will be
+registered with weaveDNS as db.weave.local. This will be in the local
+domain of our webapp. When rails connects to db, weaveDNS will take
+care of resolving that address to our postgres container.
+
+Keep in mind, that in this example, we're not concerned about
+preserving our data with backups. However, the postgres container
+we're using will store its data at /var/lib/postgresql/data, which is
+good enough to preserve data across container restarts.
+
+## Run Database Migrations ##
+
+With our database running, we can create and migrate our rails
+database using rake.
+
+```bash
+$ docker run -it --rm -w /usr/src/app webapp rake db:create
+```
+
+## Launching Rails ##
+
+Run the first server container for our webapp:
+
+```bash
+$ docker run --name webapp-1 -p 3000:3000 -d webapp
+```
+
+By giving the webapp container a unique name, it will be registered with
+weaveDNS at webapp-1.weave.local. If we wished to do advanced
+load-balancing with HAProxy or nginx, we could use these dns entries
+to route traffic. In the simple case, we could use weaveDNS for
+load-balancing between our rails containers. However, both of those
```

Comment:
Suggest either pointing at specific guides which have these features, or leave them out of this one.  This guide is already quite complicated.